sdk/src/main/java/com/bugsnag/android/User.java

```diff
@@ -1,28 +1,23 @@
 package com.bugsnag.android;
 
+import android.content.SharedPreferences;
 import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 
 import java.io.IOException;
 
 /**
  * Information about the current user of your application.
  */
 class User implements JsonStream.Streamable {
-    private String id;
-    private String email;
-    private String name;
+    @Nullable private String id;
+    @Nullable private String email;
+    @Nullable private String name;
 
-    User() {
-    }
-
-    User(String id, String email, String name) {
```

Comment:
I think for now it's worth keeping this as a public convenience constructor that builds a Builder.

Reply:
I disagree. This was never public, so why would we want to introduce something new, "for now", that would be deprecated along with the other APIs?

Exposing only the builder to create `User` objects is what will make it easier to add additional user fields in the future, while keeping the API clean.